diff --git a/node_modules/konva/lib/Context.d.ts b/node_modules/konva/lib/Context.d.ts
index 04f8f3c..204575c 100644
--- a/node_modules/konva/lib/Context.d.ts
+++ b/node_modules/konva/lib/Context.d.ts
@@ -44,6 +44,7 @@ export declare class Context {
     strokeRect(x: number, y: number, width: number, height: number): void;
     fillText(text: string, x: number, y: number, maxWidth?: number): void;
     measureText(text: string): TextMetrics;
+    getImageDataBraveCompatible(a0: number, a1: number, a2: number, a3: number): ImageData;
     getImageData(a0: number, a1: number, a2: number, a3: number): ImageData;
     lineTo(a0: number, a1: number): void;
     moveTo(a0: number, a1: number): void;
diff --git a/node_modules/konva/lib/Context.js b/node_modules/konva/lib/Context.js
index 574ed4f..c17e834 100644
--- a/node_modules/konva/lib/Context.js
+++ b/node_modules/konva/lib/Context.js
@@ -263,8 +263,19 @@ class Context {
     measureText(text) {
         return this._context.measureText(text);
     }
+    getImageDataBraveCompatible(a0, a1, a2, a3) {
+        const canvas = document.createElement('canvas');
+        canvas.width = a2;
+        canvas.height = a3;
+        const context = canvas.getContext('2d');
+
+        const layer = this;
+        context.drawImage(layer.getCanvas()._canvas, a0, a1, a2, a3, 0, 0, a2, a3);
+
+        return context.getImageData(0, 0, a2, a3);
+    }
     getImageData(a0, a1, a2, a3) {
-        return this._context.getImageData(a0, a1, a2, a3);
+        return this.getImageDataBraveCompatible(a0, a1, a2, a3)
     }
     lineTo(a0, a1) {
         this._context.lineTo(a0, a1);
